include/emp/prefab/Card.hpp

```diff
@@ -27,29 +89,50 @@ namespace prefab {
 
   public:
     /**
-     * @param state indicate whether card should be STAITC, INIT_OPEN, or INIT_CLOSED (default STAITC)
+     * @param state indicate whether card should be STATIC, INIT_OPEN, or INIT_CLOSED (default STATIC)
      * @param show_glyphs should toggle icons show in collapsible card header? (default true)
      * @param id user defined ID for card Div, (default emscripten generated)
      */
     Card(
       const std::string & state="STATIC",
       const bool & show_glyphs=true,
       const std::string & id=""
-    ): web::Div(id) {
+    ) : Card(state, show_glyphs, new internal::CardInfo(id)) { ; }
+
+  protected:
+    /**
+     * A protected constructor for a Card.
+     * @param state indicate whether card should be STATIC, INIT_OPEN, or INIT_CLOSED (default STATIC)
+     * @param show_glyphs should toggle icons show in collapsible card header? (default true)
+     * @param info_ref a pointer to the underlying ReadoutPanelInfo object for this ReadoutPanel
+     * or a pointer to a derived info object (simulating inheritance)
+     */
+    Card(
+      const std::string & state,
+      const bool & show_glyphs,
+      internal::CardInfo * info_ref
+    ) : Div(info_ref) {
 
       AddBootstrap();
       if (state == "STATIC") { // static card with no toggle enabled
-        ((emp::web::Div) *this) << card_header; // Div cast prevents trying to add to body
-        ((emp::web::Div) *this) << card_body;
+        static_cast<Div>(*this) << card_header; // Div cast prevents trying to add to body
+        static_cast<Div>(*this) << card_body;
       } else {
         // card is collapsible, make the collapse link the head of the card
         prefab::CollapseCoupling accordion(card_header,
           card_body,
           state == "INIT_OPEN",
           emp::to_string(card_base+ "_card_collapse")
         );
-        ((emp::web::Div) *this) << accordion.GetControllerDiv();
-        ((emp::web::Div) *this) << accordion.GetTargetDiv();
+
+        Div header_div{accordion.GetControllerDiv()};
+        static_cast<Div>(*this) << header_div;
+        static_cast<Div>(*this) << accordion.GetTargetDiv();
+
+        on_toggle_fun_t & tog = GetToggleHandler();
+        header_div.OnClick([header_div, &toggle = tog](){
+          toggle(/* TODO: toggle state boolean */);
```

Comment:
We should explicitly say that it’s currently unimplemented because of double-click issues